dataland-backend/src/test/kotlin/org/dataland/datalandbackend/service/DataManagerTest.kt

```diff
@@ -66,21 +55,95 @@ class DataManagerTest(
 
     @Test
     fun `add all companies and search for them one by one by using their names`() {
+        addAllCompanies(testCompanyList)
         for (company in testCompanyList) {
-            testManager.addCompany(company)
+            val searchResponse = testManager.searchCompanies(company.companyName, true)
+            assertTrue(
+                searchResponse.any { it.companyInformation.companyName == company.companyName },
+                "The posted company could not be retrieved by searching for its name."
+            )
         }
+    }
 
+    @Test
+    fun `search for identifiers and check if it can find the one`() {
+        addAllCompanies(testCompanyList)
         for (company in testCompanyList) {
-            val searchResponse = testManager.listCompaniesByName(company.companyName)
-            assertEquals(
-                company.companyName, searchResponse.first().companyInformation.companyName,
-                "The posted company could not be retrieved by searching for its name."
+            for (identifier in company.identifiers) {
+                val searchResponse = testManager.searchCompanies(identifier.identifierValue, false).toMutableList()
+                // The response list is filtered to exclude results that match in account of another identifier having
+                // the required value but the looked for identifier type does not exist (This happens due to the test
+                // data having non-unique identifier values for different identifier types)
+                searchResponse.retainAll {
+                    it.companyInformation.identifiers.any {
+                        identifierInResponse ->
+                        identifierInResponse.identifierType == identifier.identifierType
+                    }
+                }
+                assertTrue(
+                    searchResponse.all { it.companyInformation.identifiers.contains(identifier) },
+                    "The posted company could not be retrieved by searching for its identifier."
+                )
+            }
+        }
+    }
+
+    @Test
+    fun `add all companies and verify that searching for stock indices returns correct results`() {
+        val stockIndiciesInTestData = mutableSetOf<CompanyInformation.StockIndex>()
+        for (company in testCompanyList) {
+            testManager.addCompany(company)
+            stockIndiciesInTestData += company.indices
+        }
+
+        for (stockIndex in stockIndiciesInTestData) {
+            val searchResponse = testManager.searchCompaniesByIndex(stockIndex)
+            assertTrue(
+                searchResponse.all { it.companyInformation.indices.contains(stockIndex) },
+                "The posted company could not be retrieved by searching for its stock indices."
```

Comment:
Die Error Message passt finde ich nicht ganz. Sie müsste eher sowas aussagen wie:
"The search result for the stock index $stockIndex contains at least one company which is not listed in $stockIndex."

Reply:
geändert